src/Payments/Api.php

```diff
@@ -135,6 +135,15 @@ public function process_legacy_payment( PaymentContext $context, PaymentResult &
 			return;
 		}
 
+		// if we have savedPaymentMethodToken in the request, then convert into
+		// the format expected by the post.
+		// phpcs:ignore WordPress.Security.NonceVerification
+		if ( isset( $_POST['savedpaymentmethodtoken'] ) ) {
+			$token_key = 'wc-' . $context->payment_method . '-payment-token';
+			// phpcs:ignore WordPress.Security.NonceVerification, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress . Security . ValidatedSanitizedInput . InputNotSanitized
+			$_POST[ $token_key ] = wc_clean( $_POST['savedpaymentmethodtoken'] );
```

Comment:
Does this have to happen on the server – I'm wondering if we can convert it to `wc-stripe-payment-token` on the client. That way the API is consistent, there's not two different POST params related to this. i.e. tighten it up

Reply:
ya good catch, definitely could do it on the client and it would remove the need for this code 👍